include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -711,15 +711,13 @@ typedef struct OrtTensorRTProviderOptions {
 typedef struct OrtMIGraphXProviderOptions {
```

Comment:
This is a breaking change.  ONNX Runtime maintains ABI backwards compatibility. Certainly, this change breaks that.  As it will only impact the MIGraphX EP, I could accept this change. So, ORT's ABI backwards compatibility promise becomes if ORT provides ABI backwards compatibility if you don't use the functions that are not backwards compatible. (A little bit funny)
 
The correct way of doing this is to move the whole data structure out of this header file and make it opaque(invisible), and use getters/setters to access the fields. 

https://community.kde.org/Policies/Binary_Compatibility_Issues_With_C%2B%2B

Reply:
It is part of ORT's public API that everyone uses, no matter if MIGRAPHX EP is used or not.

Reply:
Can we pad this instead if this is a concern then? We're already using this internally for our builds and release, adding two padding vars should keep things consistent since we're missing a `char*` and `int` no?